dask/blockwise.py

```diff
@@ -260,11 +260,14 @@ def cull(self, keys, all_hlg_keys):
         _ = self._dict  # trigger materialization
         return self._cached_dict["basic_layer"].cull(keys, all_hlg_keys)
 
-    def map_tasks(self, func):
+    def map_tasks(self, func, pass_key=False):
         new_indices = []
         for key, input_indices in self.indices:
             if input_indices is None:  # A literal
-                new_indices.append((func([key]), None))
+                if pass_key:
+                    new_indices.append((func([key], key), None))
```

Comment:
I don't think this is strictly correct. `key` here is a literal and not a graph key. See https://github.com/dask/dask/blob/97c2f060d5ab31d831d20042e0a3e41ede3d2a64/dask/blockwise.py#L380-L382